### PR TITLE
moved all promises syntactic to async/await

### PR DESCRIPTION
**Pull request by SaurabhXRT**
This pull request addresses issue #262 by refactoring the code to use async/await syntax instead of promises where applicable. All asynchronous operations have been updated to use async/await for improved readability and maintainability.

Changes:

Converted all promise-based syntax to async/await in the affected files.
Verified that the code behaves as expected after the changes.
Related Issue:

#262

## Summary by CodeRabbit

- **Refactor**
    - Updated the `ContributorsCard` component to utilize async/await for better error handling and state management.
    - Converted function declarations to arrow functions in the `Dashboard` components to streamline component definitions.
    - Enhanced the `TemplateCard` component's `copyToClipboard` function with async/await for improved error handling and user feedback.
    - Modified server initialization and template generation functions to async/await for more efficient asynchronous operations.

### CODE REVIEW

**Comment by shravan20** [comment]
@SaurabhXRT : Nice work, will do final review over the weekend and have it merged to master. 👏🏻

File: frontend/src/components/organisms/TemplateCard/index.js [hunk 1]
```diff
@@ -43,19 +43,22 @@ const TemplateCard = (props) => {
...
-        });
+  const copyToClipboard = async () => {
+    try {
+      if (navigator.clipboard) {
+        await navigator.clipboard.writeText("![Quote](" + quoteUrl + ")");
+        setSnackbarMessage("Copied to Clipboard!");
+        setShowSnackbar(true);
+      } else {
+        throw new Error("Clipboard API not available");
+      }
```
Comment (shravan20) [inline frontend/src/components/organisms/TemplateCard/index.js:54]: Can we invert this logic?

```js
if(!navigator.clipboard) {
   throw new Error("Clipboard API not available");
}

await navigator.clipboard.writeText("![Quote](" + quoteUrl + ")");
        setSnackbarMessage("Copied to Clipboard!");
        setShowSnackbar(true);
```

We bring if else to simple if.

**Review by shravan20 — CHANGES_REQUESTED** [review]
PR LGTM, minor changes suggested.

Apologies for delayed PR review

**Comment by SaurabhXRT** [comment]
why this PR is not merged yet ..is there anything else which I have to make change?

**Review by shravan20 — APPROVED** [review]
PR LGTM

**Pull request merged**